### PR TITLE
Calculate fees

### PR DESCRIPTION
**Pull request by michalsmiarowski**
Adds a method to `TBTC` class that calculates `Bitcoin Miner Fee` and `Threshold Network Fee` which are displayed before the reveal.

We display two decimal places for TBTC amount and six decimal places for the fees

### CODE REVIEW

File: src/threshold-ts/tbtc/index.ts [hunk 1]
```diff
@@ -203,6 +221,51 @@ export class TBTC implements ITBTC {
...
+    const treasuryFee = BigNumber.from(depositTreasuryFeeDivisor).gt(0)
+      ? BigNumber.from(depositAmount).div(depositTreasuryFeeDivisor).toString()
+      : "0"
+
+    const amountToMint = BigNumber.from(depositAmount)
+      .sub(treasuryFee)
+      .toString()
+
+    // https://github.com/keep-network/tbtc-v2/blob/main/solidity/contracts/vault/TBTCOptimisticMinting.sol#L328-L336
+    const optimisticMintFee = BigNumber.from(optimisticMintingFeeDivisor[0]).gt(
```
Comment (r-czajkowski) [inline src/threshold-ts/tbtc/index.ts:255]: Let's define `optimisticMintingFeeDivisor`. I mean:
```ts
const [depositParams, _optimisticMintingFeeDivisor] =
  await this._multicall.aggregate(calls)
 
 const optimisticMintingFeeDivisor = _optimisticMintingFeeDivisor[0]
```

File: src/threshold-ts/tbtc/index.ts [hunk 2]
```diff
@@ -203,6 +221,51 @@ export class TBTC implements ITBTC {
...
+    )
+      ? BigNumber.from(amountToMint)
+          .div(optimisticMintingFeeDivisor[0])
+          .toString()
+      : "0"
+
+    return {
+      treasuryFee: treasuryFee,
+      optimisticMintFee: optimisticMintFee,
+    }
```
Comment (r-czajkowski) [inline src/threshold-ts/tbtc/index.ts:266]: We can just:
```ts
return {
 treasuryFee,
 optimisticMintFee
}
```

File: src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx [hunk 3]
```diff
@@ -27,15 +26,19 @@ const TransactionDetailsTable = () => {
...
             {bitcoinMinerFee} BTC
           </BodySm>
         </Skeleton>
       </HStack>
       <HStack justifyContent="space-between" alignItems="center">
         <BodySm color="gray.500">Threshold Network Fee</BodySm>
-        <BodySm color="gray.700">{thresholdNetworkFee} BTC</BodySm>
+        <Skeleton isLoaded={!!thresholdNetworkFee}>
+          <BodySm color="gray.700" display="flex" alignItems="center">
+            {thresholdNetworkFee} BTC
```
Comment (r-czajkowski) [inline src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx:39]: Shouldn't we format amount? `formatTokenAmount(thresholdNetworkFee, undefined, 8)`

File: src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx [hunk 4]
```diff
@@ -6,18 +6,17 @@ import shortenAddress from "../../../../utils/shortenAddress"
...
     ethAddress,
   } = useTbtcState()
+
   return (
     <Stack spacing={2} mb={6}>
       <HStack justifyContent="space-between" alignItems="center">
         <BodySm color="gray.500">Minted Amount</BodySm>
-        <Skeleton isLoaded={!isLoadingTbtcMintAmount}>
+        <Skeleton isLoaded={!!tBTCMintAmount}>
           <BodySm color="gray.700">{tBTCMintAmount} tBTC</BodySm>
```
Comment (r-czajkowski) [inline src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx:20]: Same as above

File: src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx [hunk 5]
```diff
@@ -27,15 +26,19 @@ const TransactionDetailsTable = () => {
       </HStack>
       <HStack justifyContent="space-between">
         <BodySm color="gray.500">Bitcoin Miner Fee</BodySm>
-        <Skeleton isLoaded={!isLoadingBitcoinMinerFee}>
+        <Skeleton isLoaded={!!bitcoinMinerFee}>
           <BodySm color="gray.700" display="flex" alignItems="center">
             {bitcoinMinerFee} BTC
```
Comment (r-czajkowski) [inline src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx:31]: Same as above.

File: src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx [hunk 6]
```diff
@@ -22,6 +22,7 @@ import { QRCode } from "../../../../components/QRCode"
 import { useThreshold } from "../../../../contexts/ThresholdContext"
 import { UnspentTransactionOutput } from "@keep-network/tbtc-v2.ts/dist/bitcoin"
 import withOnlyConnectedWallet from "../../../../components/withOnlyConnectedWallet"
+import { BigNumber } from "ethers"
```
Comment (r-czajkowski) [inline src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx:25]: Unnecessary import ?

**Review by r-czajkowski — CHANGES_REQUESTED** [review]

File: src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx (hunk 3 above)
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx]: 08e07daa00b11a0fdc199c27925772c788d224ae

File: src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx (hunk 4 above)
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx]: 08e07daa00b11a0fdc199c27925772c788d224ae

File: src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx (hunk 5 above)
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/components/TransactionDetailsTable.tsx]: 08e07daa00b11a0fdc199c27925772c788d224ae

File: src/threshold-ts/tbtc/index.ts (hunk 1 above)
Comment (michalsmiarowski) [reply src/threshold-ts/tbtc/index.ts]: 6318d78632feab0be3845838021bd87bea242338

File: src/threshold-ts/tbtc/index.ts (hunk 2 above)
Comment (michalsmiarowski) [reply src/threshold-ts/tbtc/index.ts]: a202213aef3d71babd56bdec871b3997c448d5f2

File: src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx (hunk 6 above)
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx]: 632df3a425ebe3e2ebe11df388ed75302545ca6d

**Review by r-czajkowski — APPROVED** [review]

**Pull request merged**